src/Analyzer.Core/Rules/BuiltInRules.json

```diff
@@ -633,6 +633,125 @@
       ]
     }
   },
+  {
+    "name": "Kubernetes_Upgrade_Non-Vulnerable_Version",
+    "description": "Kubernetes Services should be upgraded to a non-vulnerable Kubernetes version",
+    "recommendation": "Upgrade to a non-vulnerable Kubernetes version (1.11.9+, 1.12.7+, 1.13.5+, and 1.14.0+)",
+    "helpUri": "https://github.com/Azure/template-analyzer/blob/main/docs/built-in-bpa-rules.md/#kubernetes-services-should-be-upgraded-to-a-non-vulnerable-kubernetes-version",
+    "evaluation": {
+      "resourceType": "Microsoft.ContainerService/managedClusters",
+      "allOf": [
+        {
+          "not": {
+            "path": "properties.kubernetesVersion",
+            "in": [
+              "1.13.4",
+              "1.13.3",
+              "1.13.2",
+              "1.13.1",
+              "1.13.0"
+            ]
+          }
+        },
+        {
+          "not": {
+            "path": "properties.kubernetesVersion",
+            "in": [
+              "1.12.6",
+              "1.12.5",
+              "1.12.4",
+              "1.12.3",
+              "1.12.2",
+              "1.12.1",
+              "1.12.0"
+            ]
+          }
+        },
+        {
+          "not": {
+            "path": "properties.kubernetesVersion",
+            "in": [
+              "1.11.8",
+              "1.11.7",
+              "1.11.6",
+              "1.11.5",
+              "1.11.4",
+              "1.11.3",
+              "1.11.2",
+              "1.11.1",
+              "1.11.0"
+            ]
+          }
+        },
+        {
+          "not": {
+            "path": "properties.kubernetesVersion",
```

Comment:
Can't we collapse all these last not properties with a regex like:
```
^1.[0-9].*
```
And maybe add another for the number 10, or:
```
^1.[0|1|2|3|4|5|6|7|8|9|10].*
```

Reply:
Wait, in fact we can collapse all of them with:

```
{
  "not": {
    "path": "properties.kubernetesVersion",
    "regex": ""^1.((([0-9]|10).*)|(11.[0-8])|(12.[0-6])|(13.[0-4]))"
  }
}
```

Reply:
I was thinking about this, but for readability purposes I decided to leave it as is. I'm open to changing it though.

Reply:
I like the regex better because for a programmer it shouldn't be that hard, it's just a couple of ranges and a couple of disjunctions, especially if you put it in an editor that shows you the relationship between the parenthesis and stuff like that (I used https://regex101.com/ to write it :)). And for a non-programmer, I feel that the rule description is clear enough that they shouldn't need to look at the code

Reply:
When I wrote a prototype for this a while back (during rule language creation) I did it in a single regex.  I do like the clarity and readability of separating it out though.  I'm ok with either method, but a hybrid might bring the best of both (clarity and compactness), like a regex for each minor version after 1.10, and one for everything <10?

Reply:
You're suggesting having two regex patterns, one for the ones that are listed in the list form and another for the ones that are currently represented as regex?

Reply:
@yane3628 Based off how I read I think he meant four: for <=1.10, 1.11, 1.12, and 1.13. However, I would vote for the two like you said, which is just taking Vera's regex and basically splitting it in half:
^1.(([0-9]|10).*)
^1.((11.[0-8])|(12.[0-6])|(13.[0-4]))

As far as regexes go, these are fairly legible to me.

Reply:
Any of these sound good to me :)

Reply:
Updated it. I will also add that by having them all separated, it is much clearer what the tests should look like.